bindings/java/Makefile

```diff
@@ -18,7 +20,16 @@ test: build
 gradlew:
 	gradle --no-daemon setup
 
-$(OUT_DIR)/lib/libevmc-java.so:
+docker-linux: docker/linux.Dockerfile
```

Comment:
Trying to understand the need for a docker-linux rule in the Makefile.  iirc, the CI job uses a linux container to build.   I'm guessing you were developing on MacOs and needed a way to generate .so files for linux? 

Is the end goal here to provide support for MacOs and Linux?  If so, instead of Making an explicit docker-linux rule...I would detect the OS in the Makefile and modify a variable that contains the flags/includes/paths in a conditional and leave the rule alone.  I think the original PR to add Java-Bindings had this feature, but I imagine it was removed when getting it to work with the CMAKE build process.  

@axic / @chfast can @atoulme conform to your build process and add support for producing shared libs for different platforms (Linux, MacOs, Windows, etc)?